### DISCUSSION TITLE
specify the percentage of matching

### DISCUSSION THREAD

**Discussion by SehamAAlharbi** [jplag/JPlag#781] [Q&A] [answered]
Hello, 

I am currently using JPlag API to find the similarity among a large set of java files. I was wondering whether there is a way of limiting certain matches with e.g., 100% or 0% of matching from the generated report? and matching files? 

Thanks in advance.
Seham

**Comment by tsaglam** [comment, marked as answer]
For a lower bound, you can use the following CLI argument:
```
 -m M                   Comparison similarity threshold [0.0-1.0]:  All  comparisons  above this threshold will
                         be saved (default: 0.0)
```
For example, `-m 0.5` will filter all submissions with a similarity of 50% or less. We currently do not have a mechanism to filter submissions when their similarity is _above_ a certain threshold.

You can also enforce some restrictions on certain files before the comparison:

- The argument `-s name` limits the comparison to only the files residing in the "name" folder of each submission.
- The argument `-x  path` allows specifying a text file that contains file names to ignore.
